TreeNode: add node class callback function

jiankian opened the pull request:
Tree树形组件，新增类似Vue3版本的自定义树形节点class类名回调方法，方便实现自定义的Tree树形组件样式

Please make sure these boxes are checked before submitting your PR, thank you!

* [x] Make sure you follow Element's contributing guide ([中文](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.zh-CN.md) | [English](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.en-US.md) | [Español](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.fr-FR.md)).
* [x] Make sure you are merging your commits to `dev` branch.
* [x] Add some descriptions and refer relative issues for you PR.

jiankian committed:
TreeNode: add node class callback function
Tree树形组件，新增类似Vue3版本的自定义树形节点class类名回调方法，方便实现自定义的Tree树形组件样式

wangdaodao commented on packages/tree/src/tree.vue:
```diff
@@ -17,6 +17,7 @@
       :show-checkbox="showCheckbox"
       :key="getNodeKey(child)"
       :render-content="renderContent"
+      :tree-node-class="treeNodeClass"
```
`tree-node.vue` 中的 `el-tree-node` 标签应该也增加 `:tree-node-class="treeNodeClass"` 。

jiankian replied:
对的

jiankian replied:
大意了，加了才是完整的

jiankian committed:
TreeNode: add node class callback function
Tree树形组件，新增类似Vue3版本的自定义树形节点class类名回调方法，方便实现自定义的Tree树形组件样式

jiankian commented:
请求合并然后更新到官方库中，不然自定义树形结构组件还得自己拷贝一份项目代码处理
`tree-node.vue` 中的 `el-tree-node` 标签也已增加` :tree-node-class="treeNodeClass"`

jiankian committed:
修正项目运行和打包错误问题：
node-sass版本太老，只支持node v14版本问题
修正老版本node-sass不支持Windows 11环境问题
fix build problems:
fix node-sass version to support node v14 - v16, node-sass v4 is just support node v14
fix sass-loader support for node-sass v6
fix node-sass not support Windows 11
Error: Node Sass does not yet support your current environment: Windows 64-bit with Unsupported runtime (83)

jiankian committed:
webpack 打包器 依赖版本升级

jiankian committed:
vue 版本和loader版本升级
